pkg/agent/prommetrics.go

```diff
@@ -115,6 +163,9 @@ func makePrometheusParts(globalstate *agentState) (PromMetrics, *prometheus.Regi
 		schedulerRequests:         schedulerRequests,
 		informantRequestsOutbound: informantRequestsOutbound,
 		informantRequestsInbound:  informantRequestsInbound,
+		runnerFatalErrors:         runnerFatalErrors,
 		runnerThreadPanics:        runnerThreadPanics,
+		runnerStarts:              runnerStarts,
+		runnerRestarts:            runnerRestarts,
```

Comment:
I think having to edit 2-4 spots to add a metric, it might be cleaner to declare `PromMetrics` at the top of the file, set the fields on this when you construct the runner, and then call `MustRegister` for each metric.

Reply:
I think I agree this is an issue, but I don't think it can get much better than it is now (unless I'm missing something?). To me it seems like we'd kind of always have to have multiple places — the struct definition of `PromMetrics`, instantiating `PromMetrics`, and `MustRegister`-ing the metric. (plus whatever usage is needed throughout the codebase)

Given that, I'm inclined to leave it as is. Thoughst?

Reply:
Yeah, this will be resolved post-merge by #276.